test/integration/vm_translator_test.rb

```diff
@@ -1,39 +1,14 @@
 require "test_helper"
 
 class VMTranslatorIntegrationTest < Minitest::Test
-  def test_integration_test
-    assembly_code = `bin/vm-translator examples/Push.vm`
-    expected = File.read("test/fixtures/Push.asm")
-    assert_equal(expected, assembly_code)
-  end
-
-  def test_integration_test_simple_add
-    assembly_code = `bin/vm-translator examples/SimpleAdd.vm`
-    expected = File.read("test/fixtures/SimpleAdd.asm")
-    assert_equal(expected, assembly_code)
-  end
-
-  def test_integration_test_simple_eq
-    assembly_code = `bin/vm-translator examples/SimpleEq.vm`
-    expected = File.read("test/fixtures/SimpleEq.asm")
-    assert_equal(expected, assembly_code)
-  end
-
-  def test_integration_test_stack_test
-    assembly_code = `bin/vm-translator examples/StackTest.vm`
-    expected = File.read("test/fixtures/StackTest.asm")
-    assert_equal(expected, assembly_code)
-  end
-
-  def test_integration_test_basic_test
-    assembly_code = `bin/vm-translator examples/BasicTest.vm`
-    expected = File.read("test/fixtures/BasicTest.asm")
-    assert_equal(expected, assembly_code)
-  end
-
-  def test_integration_test_pointer_test
-    assembly_code = `bin/vm-translator examples/PointerTest.vm`
-    expected = File.read("test/fixtures/PointerTest.asm")
-    assert_equal(expected, assembly_code)
+  Dir.glob("examples/**").each do |file_path|
```

Comment:
If we’re going to use [`Dir.glob`](https://rubydoc.info/stdlib/core/Dir.glob) here (versus e.g. the simpler [`Dir.each_child(…)`](https://rubydoc.info/stdlib/core/Dir.each_child) or [`Dir.new(…).each_child`](https://rubydoc.info/stdlib/core/Dir:each_child)) then we should take more advantage of its power. Two ideas come to mind:

* if we use `*.vm` instead of `**` in the pattern, it’ll explicitly select only files with a `.vm` extension, which more clearly provides the guarantee that the subsequent lines of code assume (although I admit it doesn’t make any practical difference right now!); and
* if we pass a `base: "examples"` argument instead of including `examples/` in the pattern, the resulting paths will be relative to that `base` directory (e.g. `"BasicTest.vm"`) rather than including it as a prefix (e.g. `"examples/BasicTest.vm"`), which will save us the work of removing that prefix later.